
# How to find restriction enzyme frequency in whole genome and count the distance between them?










                        This question was migrated from Stack Overflow because it can be answered on Bioinformatics Stack Exchange.
                        Migrated 1 hour ago.
                    





I am having possibly a simple question but its huge difficulty for me.
My objective is to take the reference FASTA file and another file with 1500 restriction enzymes with its cutting sites (mostly pattern I would say), now I first want to see that how many times each enzyme cut the FASTA file and second to check the number and size of fragment that each enzyme would give me, this output pregerably would be a CSV file may be.
I tried to do pattern search using R, but its just collapsing my system. I am looking for bash command may be.

        