
# Convert fasta file (.fasta) to count file (.cf)

I am trying to convert a fasta file (.fasta) to a count file (.cf) to be able to upload it to IQTREE for use in phylogenetic analysis. The fasta file is a merged file of an alignment and a mapping of the individuals sequenced into one of 5 populations. A sample of this file has been attached.
I have tried both R and python but am not having any success.

        